lib/common/nvpair.c

```diff
@@ -946,7 +946,7 @@ pcmk__xe_get_bool_attr(const xmlNode *node, const char *name, bool *value)
         *value = ret;
         return pcmk_rc_ok;
     } else {
-        return pcmk_rc_unknown_format;
+        return EINVAL;
```

Comment:
The reason we initially didn't use EINVAL was because it gets mapped to the CRM_EX_SOFTWARE exit status ("Internal software bug") but this would typically be an invalid user-supplied string. Maybe pcmk_rc_bad_nvpair aka "Bad name/value pair given" (-> CRM_EX_INVALID_PARAM aka "Invalid parameter") or a new code for "not a valid boolean value"

Reply:
For pcmk__xe_get_bool_attr at least, we are already returning EINVAL if name or value are NULL.